dissect/target/plugins/os/windows/regf/appxdebugkeys.py

```diff
@@ -0,0 +1,122 @@
+from typing import Iterator
+
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.helpers.descriptor_extensions import (
+    RegistryRecordDescriptorExtension,
+    UserRecordDescriptorExtension,
+)
+from dissect.target.helpers.record import create_extended_descriptor
+from dissect.target.helpers.regutil import RegistryKey, RegistryKeyNotFoundError
+from dissect.target.plugin import Plugin, export
+
+AppxDebugKeyRecord = create_extended_descriptor([RegistryRecordDescriptorExtension, UserRecordDescriptorExtension])(
+    "windows/registry/appxdebug/key",
+    [
+        ("datetime", "ts"),
+        ("string", "name"),
+        ("string", "debug_info"),
+    ],
+)
+
+
+class AppxDebugKeysPlugin(Plugin):
+    """Plugin that iterates various AppX debug key locations."""
+
+    def _walk(self, key: RegistryKey) -> Iterator[AppxDebugKeyRecord]:
+        user = self.target.registry.get_user(key)
+        values = False
+        for value in key.values():
+            values = True
+            yield AppxDebugKeyRecord(
+                ts=key.ts,
+                name=value.name,
+                debug_info=value.value,
+                _target=self.target,
+                _key=key,
+                _user=user,
+            )
+
+        for subkey in key.subkeys():
+            values = True
+            yield from self._walk(subkey)
+
+        if not values:
+            yield AppxDebugKeyRecord(
+                ts=key.ts,
+                name=None,
+                debug_info=None,
+                _target=self.target,
+                _key=key,
+                _user=user,
+            )
+
+    def _packaged_appx_debug_keys(self) -> Iterator[AppxDebugKeyRecord]:
+        # "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\PackagedAppXDebug\\*",
+        # the * are AppX pacakge names
+        # value_name="(Default Value)", value_data=<debugger cmd>
+        key_path = "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\PackagedAppXDebug"
+
+        for key in self.target.registry.keys(key_path):
+            for subkey in key.subkeys():
+                yield from self._walk(subkey)
+
+    def _activatable_classes_debug_keys(self) -> Iterator[AppxDebugKeyRecord]:
+        # "HKEY_CURRENT_USER\\Software\\Classes\\ActivatableClasses\Package\\*\\DebugInformation\\*",
+        # the 1st * are AppX pacakge names
+        # the 2nd * look like AppX package components
+        # value_name="DebugPath", value_data=<debugger cmd>
+        key_path = "HKEY_CURRENT_USER\\Software\\Classes\\ActivatableClasses\\Package"
```

Comment:
Wouldn't it be nicer to turn this into a class constant? Like REGKEY_ACTIVATABLE_CLASS_PACKAGE for example?